### ISSUE TITLE
Opening already opened file with io.open() returns IOError.

### ISSUE THREAD

**Issue by n-kovacevic** [IronLanguages/main#1611]
Trying to open file for reading that is already in use by other process results in IOError exception. 
This happens only when using open method from io module, not when using built-in open method.

Example in IronPython 2.7.5 and 2.7.7:
```
>>> import io
>>> with io.open(r"C:\Users\nkovacevic\Desktop\test.zip", mode="rb") as f:
...     f.read()
... 
Traceback (most recent call last):
  File "<string>", line 1, in <module>
IOError: [Errno 32] The process cannot access the file 'C:\Users\nkovacevic\Desktop\test.zip' because it is being used by another process.: C:\Users\nkovacevic\Desktop\test.zip
```
The same code in CPython 2.7.13 doesn't raise exception:
```
>>> import io
>>> with io.open(r'C:\Users\nkovacevic\Desktop\test.zip', mode="rb") as f:
...     f.read()
... 
'PK\x03\x04\n\x00\x00\x00\x00\x00\xb0tpJ\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x15\x00\x00\x00New Text Document.txtPK\x01\x02?\x00\n\x00\x00\x00\x00\x00\xb0tpJ\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x15\x00$\x00\x00\x00\x00\x00\x00\x00 \x00\x00\x00\x00\x00\x00\x00New Text Document.txt\n\x00 \x00\x00\x00\x00\x00\x01\x00\x18\x00\xbb*\x8buZ\x9e\xd2\x01\xbb*\x8buZ\x9e\xd2\x01\xbb*\x8buZ\x9e\xd2\x01PK\x05\x06\x00\x00\x00\x00\x01\x00\x01\x00g\x00\x00\x003\x00\x00\x00\x00\x00'
>>> 
```
Main problem with this is that **pathlib** library also uses io.open and not built-in open method.

**Comment by slozier** [comment]
@slide I believe this one was fixed with #1586?

**Comment by slide** [comment]
Yes, I think it should be.
